processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationFilterUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * processor-api
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.processor.api.filter;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import com.synopsys.integration.alert.common.persistence.model.job.BlackDuckProjectDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.FilteredDistributionJobResponseModel;
+import com.synopsys.integration.alert.processor.api.detail.DetailedNotificationContent;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+public class JobNotificationFilterUtils {
+    public static boolean doesNotificationApplyToJob(FilteredDistributionJobResponseModel filteredDistributionJobResponseModel, DetailedNotificationContent detailedNotificationContent) {
+        String notificationType = detailedNotificationContent.getNotificationContentWrapper().extractNotificationType();
+        NotificationType notificationTypeEnum = NotificationType.valueOf(notificationType);
+
+        if (!doesNotificationTypeMatch(filteredDistributionJobResponseModel, notificationType)) {
+            return false;
+        }
+        String projectName = detailedNotificationContent.getProjectName().orElse("");
+        if (!doesProjectApplyToJob(filteredDistributionJobResponseModel, projectName)) {
+            return false;
+        }
+        switch (notificationTypeEnum) {
+            case VULNERABILITY:
+                List<String> notificationSeverities = detailedNotificationContent.getVulnerabilitySeverities();
+                if (!doVulnerabilitySeveritiesApplyToJob(filteredDistributionJobResponseModel, notificationSeverities)) {
+                    return false;
+                }
+                break;
+            case POLICY_OVERRIDE:
+            case RULE_VIOLATION:
+            case RULE_VIOLATION_CLEARED:
+                String policyName = detailedNotificationContent.getPolicyName().orElse("");
+                if (!doesPolicyApplyToJob(filteredDistributionJobResponseModel, policyName)) {
+                    return false;
+                }
+                break;
```

Comment:
Can we return `doesPolicyApplyToJob(filteredDistributionJobResponseModel, policyName)`?